packages/cli/src/ceramic-daemon.ts

```diff
@@ -524,36 +517,111 @@ export class CeramicDaemon {
     }
   }
 
-  async addModelsToIndex(req: Request, res: Response): Promise<void> {
-    const { modelIDStrings, error } = this._validateModelIDStrings(req.body.models)
-    if (error) {
-      res.status(422).json({ error: error })
+  private async _parseDidJWSAuthHeader(
+    authHeader: string | undefined
+  ): Promise<{ kid: string, requestPath: string, timestamp: number, forModels: Array<string> }> {
+    const jwsString = authHeader.split("Authorization: Basic ")[1]
+    const result = await this.ceramic.did.verifyJWS(jwsString)
+    return {
+      kid: result.kid,
+      requestPath: result.payload.requestPath,
+      timestamp: result.payload.timestamp,
+      forModels: result.payload.requestBody ? result.payload.requestBody.models : undefined,
+    }
+  }
+
+  private _compareStringArrays(left: Array<string>, right: Array<string>): boolean {
+    return left.length === right.length && left.every((value, index) => value === right[index])
+  }
+
+  private async _validateDIDJWSAuthHeader(
+    basePath: string,
+    authHeader: string | undefined,
+    forModels?: Array<string>
+  ): Promise<{ kid?: string, error?: string }> {
+    const errorResult = { status: 'error', error: 'Missing or invalid authorization signature' }
```

Comment:
Sure we can. I didn't want to do it initially, because I think that in security-related contexts it's usually better not to give out too much information that could help attackers. Happy to change this, if you're fine with this, though.

Reply:
Personally, I'd say that giving out information about the node configuration is a different than telling a user what about their specific request caused it to fail.  I don't think telling the user that they used the wrong did, or had an expired authorization gives out any extra information about the node itself (but I'm not a security expert so maybe there's something subtle here I'm missing?).  My personal instinct certainly always leans towards giving devs as much information as possible to understand when they are doing something wrong.